### DISCUSSION TITLE
PalavraChave birl lexador, BORA CUMPADE 0;

### DISCUSSION THREAD

**Discussion by ItaloCobains** [DesignLiquido/delegua#165] [P&R] [answered]
Bom dia, @samuelrvg @leonelsanchesdasilva.
Na branch **dialeto-birl**, arquivo lexador-birl.ts, linha 58, preciso fazer esse while parar com uma validação para a instrução 
```c
BORA CUMPADE 0;
```
Porém não gostaria de usar o _0_ para validar isso pelo simples motivo de que o retorno nem sempre vai ser _0_.
Para melhor entendimento essa instrução e semelhante ao 
```c
return 0;
```

Caso não acharmos um solução terei que usar um prefixo como **?** para fazer esse processo.

**Comment by leonelsanchesdasilva** [comment]
@ItaloCobains `BORA CUMPADE` é o equivalente a `retorna` em outros dialetos, portanto, o zero deve ser mapeado como um literal.

**Reply by ItaloCobains** [reply]
Exato @leonelsanchesdasilva , o Problema é que preciso pegar a frase `BORA CUMPADE` no while, e sem algo pra validar eu acabo pegando a expressão inteira.
exemplo: `BORA CUMPADE 0;`

**Reply by leonelsanchesdasilva** [reply]
@ItaloCobains Como o lexador gera as tokens pra este caso?

**Reply by ItaloCobains** [reply]
@leonelsanchesdasilva a função a baixo e responsável por isso ela percorre a código ate chegar em um _\n_ ou _?_ com isso eu faço a formatação da string e comparo no switch.

```ts
identificarPalavraChave(): void {
        while (this.simboloAtual() !== '\n' && this.simboloAtual() !== '?') {
            this.avancar();
        }

        const proximo = this.simboloAtual();

        const valor =
            proximo !== '?'
                ? this.codigo.substring(this.inicioSimbolo, this.atual - 1).trim()
                : this.codigo.substring(this.inicioSimbolo, this.atual + 1).trim();

        switch (valor) {
            case 'HORA DO SHOW':
                this.adicionarSimbolo(tiposDeSimbolos.HORA_DO_SHOW);
                this.avancar();
                break;
            case 'BIRL':
                this.adicionarSimbolo(tiposDeSimbolos.BIRL);
                this.avancar();
                break;
            case 'CE QUER VER ESSA PORRA?':
                this.adicionarSimbolo(tiposDeSimbolos.CE_QUER_VER_ESSA_PORRA);
                this.avancar();
                break;
            case 'BORA CUMPADE':
                this.adicionarSimbolo(tiposDeSimbolos.BORA_CUMPADE);
                this.avancar();
            default:
                this.avancar();
                break;
        }
    }
```

**Comment by ItaloCobains** [comment, marked as answer]
Resolvi mudar o comando para _BORA CUMPADE?_ para não complicar a validação.
